src/ui.tsx

```diff
@@ -96,9 +89,44 @@ export class UserInterface extends Component<Props, State> {
       }
     });
 
-    this.filteredMeta = filteredMeta;
-    this.setState({
-      imageNames: this.getImageNames(filteredMeta),
+    this.setState({ filteredMeta });
+  };
+
+  handleOnSortSubmit = (key: string, sortOrder: string): void => {
+    // Handle sort by any string or by date.
+    const sort = sortOrder === "asc" ? 1 : -1;
+
+    function compare(a: string | Date, b: string | Date): number {
+      if (a < b) {
+        return -1;
+      }
+      if (a > b) {
+        return 1;
+      }
+      return 0;
+    }
+
+    this.setState((prevState) => {
+      if (key.toLowerCase().includes("date")) {
```

Comment:
You could detect whether the _value_ is a date as they will be ISO strings? 

Also, maybe pass `sort` into the compare func, and do any type casting there? Will let you pass anything to the same function. 

I think in practice, dates won't be strings here, they will be actual Date objects, so we can have them as as type?


(happy to merge as is tho tbh)

Reply:
Makes sense! At the moment the metadata comes from a temporary metadata.json file we import. The way in which we store this data may change, so I'll be tempted to open a new issue and reconsider this when the data types are final?

Reply:
Ok for passing `sort` in the compare function.

Reply:
> Makes sense! At the moment the metadata comes from a temporary metadata.json file we import. The way in which we store this data may change, so I'll be tempted to open a new issue and reconsider this when the data types are final?

Yep